expression/builtin_compare.go

```diff
@@ -745,6 +752,14 @@ func (c *leastFunctionClass) getFunction(ctx sessionctx.Context, args []Expressi
 	}
 	switch tp {
 	case types.ETInt:
+		// default unsigned, if one field is signed, return signed result
```

Comment:
Can we extract a common function for greatest/least? There are too many duplicated codes

Reply:
I can try, but I am not sure if it is better, the reason is:

1. greatest/least are two fixed features, little possibility to extend the similar function
2. in this issue: greatest/least shows the different logic
3. every function has an independent class, it little weird to merge two functions into one

btw, if we really should do it, I can do this in the next pr, since the issue is a bug fix?